src/riscv/linux/init.c

```diff
@@ -509,8 +571,9 @@ void cpuinfo_riscv_linux_init(void) {
 	size_t valid_clusters_index = 0;
 	size_t valid_packages_index = 0;
 	size_t valid_uarchs_index = 0;
+	size_t l2_count = 0;
 	last_uarch = cpuinfo_uarch_unknown;
-	for (size_t processor = 0; processor < max_processor_id; processor++) {
+	for (size_t processor = 0; processor < riscv_linux_processors_count; processor++) {
 		if (!bitmask_all(riscv_linux_processors[processor].flags, CPUINFO_LINUX_FLAG_VALID)) {
```

Comment:
Does the qsort above guarantee that all the invalid processors are at the end of riscv_linux_processors?  If so, should this be a break rather than a continue?